### PR TITLE
Lightdm WIP

### PR DESCRIPTION
**Pull request by pyhalov**

### CODE REVIEW

**Comment by xen0l** [comment]
No REQURIED_PACKAGES?

File: components/desktop/lightdm/files/svc-lightdm [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+# Use is subject to license terms.
+#
+#pragma ident	"@(#)svc-gdm.src	1.1	06/02/06 SMI"
+#
+# For modifying parameters passed to dtlogin, do not edit
+# this script. Instead use svccfg(1m) to modify the SMF
+# repository. For example:
+#
+# # svccfg
+# svc:> select application/graphical-login/gdm
```
Comment (xen0l) [inline components/desktop/lightdm/files/svc-lightdm]: gdm -> lightdm

File: components/desktop/lightdm/files/svc-lightdm [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+# svc:/application/graphical-login/gdm> setprop gdm/args = "--fatal-warnings"
```
Comment (xen0l) [inline components/desktop/lightdm/files/svc-lightdm]: gdm -> lightdm

File: components/desktop/lightdm/files/svc-lightdm [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+# svc:/application/graphical-login/gdm> exit
```
Comment (xen0l) [inline components/desktop/lightdm/files/svc-lightdm]: gdm -> lightdm

File: components/desktop/lightdm/lightdm.p5m [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+file files/lightdm-session path=usr/bin/lightdm-session
+file files/lightdm.xml path=lib/svc/manifest/application/graphical-login/lightdm.xml pkg.depend.bypass-generate=.*
+file files/svc-lightdm path=lib/svc/method/svc-lightdm mode=0555
+
+group groupname="lightdm" gid="54"
+user gcos-field="GDM Reserved UID" group=lightdm home-dir=/var/lib/lightdm-data uid=54 username=lightdm
+
+dir path=var/lib/lightdm-data mode=1770 group=lightdm
+
+#file path=etc/apparmor.d/abstractions/lightdm
```
Comment (xen0l) [inline components/desktop/lightdm/lightdm.p5m:38]: Why are these commented?

File: components/desktop/lightdm/files/svc-lightdm [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+	if [ $rc != 0 ]; then
+		echo "$0: lightdm failed with $rc"
+		exit 1
+	fi
+	;;
+
+'stop')
+	PIDFILE=/var/run/lightdm.pid 
+	if test '!' -f $PIDFILE ; then
+	        echo "$PIDFILE doesn't exist, perhaps LIGHTDM isn't running"
```
Comment (xen0l) [inline components/desktop/lightdm/files/svc-lightdm]: Is LIGHTDM or LightDM?

File: components/desktop/lightdm/lightdm.p5m [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
+
+<transform file path=etc/.* -> default preserve true>
+
+file files/lightdm-session path=usr/bin/lightdm-session
+file files/lightdm.xml path=lib/svc/manifest/application/graphical-login/lightdm.xml pkg.depend.bypass-generate=.*
+file files/svc-lightdm path=lib/svc/method/svc-lightdm mode=0555
+
+group groupname="lightdm" gid="54"
+user gcos-field="GDM Reserved UID" group=lightdm home-dir=/var/lib/lightdm-data uid=54 username=lightdm
```
Comment (xen0l) [inline components/desktop/lightdm/lightdm.p5m]: GDM -> LightDM

File: components/desktop/lightdm/patches/13-save-audio-settings.patch [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
++   
++    user_name=user_get_name(user);
++    pwd=getpwnam(user_name);
++    if(pwd == NULL || pwd->pw_dir == NULL)
++       return;
++
++    if(stat("/dev/audio",&st))
++      return;
++   
++    if(st.st_uid==user_get_uid(user)) {
```
Comment (xen0l) [inline components/desktop/lightdm/patches/13-save-audio-settings.patch]: add spaces before and after == for consistency

File: components/desktop/lightdm/patches/13-save-audio-settings.patch [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
++
++       drop_privileges = geteuid () == 0;
++       if (drop_privileges)
++          privileges_drop (user_get_uid (user), user_get_gid (user));
++
++       save_dir=g_strdup_printf("%s/.audioctl", pwd->pw_dir);
++       if (!save_dir) 
++          return;
++    
++       if(stat(save_dir,&st_sd)) {
```
Comment (xen0l) [inline components/desktop/lightdm/patches/13-save-audio-settings.patch]: space after ,

File: components/desktop/lightdm/lightdm.p5m (hunk 4 above)
Comment (pyhalov) [inline components/desktop/lightdm/lightdm.p5m:38]: To remember that we shouldn't ship them

**Pull request merged**